petergreve/week1/ass2/index.html

```diff
@@ -0,0 +1,139 @@
+<html>
+<head>
+	<title>Hello world with react</title>
+	<style type="text/css">
+	body {
+		width: 1000px;
+		margin: auto;
+	}
+	</style>
+
+	<script src="http://fb.me/react-0.14.7.min.js"></script>
+	<script src="http://fb.me/react-dom-0.14.7.min.js"></script>
+	<!-- <script src="http://cdnjs.cloudflare.com/ajax/libs/babel-standalone/6.4.4/babel.min.js"></script> -->
+	<script src="http://fb.me/JSXTransformer-0.12.2.js"></script>
+</head>
+<body>
+	<div id="root"> </div>
+
+	<!-- 1. Create an inputfield that doesn't accept any numbers -->
+	<!-- 2. Find a way to submit and print the message -->
+	<!-- 3. Validation errors in inputfields should produce an error-message rather than a hello morld message -->
+	<!-- 4. Any error-message should be printed in red color -->
+	<!-- 5. Provide a function that returns all messages ever printed -->
+	<!-- 6. This must be solved by using only native Javascript and react -->
+	<!-- 7. All markup must be produced with react -->
+	<!-- 8. All messages must be printed in the order they where submitted -->
+
+
+
+<script type="text/jsx">
+
+	var InputComponent = React.createClass({
+		  render: function() {
+		    return 	<div>
+		    			<input type="text" ref="input" />
+						<button type="button" onClick={this.validate}>Submit</button>
+		    		</div>;
+		  },
+		  validate: function() {
+		  	var message = this.refs.input.getDOMNode().value;
```

Comment:
You do not need `getDOMNode()`, `this.refs.input.value` would work fine. However, as pointed in the documentation https://facebook.github.io/react/docs/more-about-refs.html#the-ref-string-attribute:

> React also supports using a string (instead of a callback) as a ref prop on any component, although this approach is mostly legacy at this point.

So you could use a callback: `<input type="text" ref={(component) => this.input = component} />` and then you have access to the input value via `this.input-value`